src/caffe/layer_factory.cpp

```diff
@@ -160,14 +160,18 @@ REGISTER_LAYER_CREATOR(TanH, GetTanHLayer);
 #ifdef WITH_PYTHON_LAYER
 template <typename Dtype>
 shared_ptr<Layer<Dtype> > GetPythonLayer(const LayerParameter& param) {
+  string module_name = param.python_param().module();
+  string layer_name = param.python_param().layer();
   Py_Initialize();
   try {
-    bp::object module = bp::import(param.python_param().module().c_str());
-    bp::object layer = module.attr(param.python_param().layer().c_str())(param);
+    bp::object globals = bp::import("__main__").attr("__dict__");
+    bp::exec(("import " + module_name).c_str(), globals, globals);
+    bp::object layer_class = bp::eval((module_name + "." + layer_name).c_str(),
```

Comment:
I could be mistaken (and/or paranoid), but isn't it dangerous to call `eval` on a string that hasn't been checked?  That is, we should check that `module_name` and `layer_name` both match the regex `^[a-zA-Z_][a-zA-Z0-9_]*$.